src/Stratis.Bitcoin.Features.PoA/PoAConsensusManagerBehavior.cs

```diff
@@ -44,7 +44,7 @@ protected override async Task OnMessageReceivedAsync(INetworkPeer peer, Incoming
 
         /// <inheritdoc />
         /// <remarks>Creates <see cref="PoAHeadersPayload"/> instead of <see cref="HeadersPayload"/> like base implementation does.</remarks>
-        protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersPayload, out ChainedHeader lastHeader)
+        public override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersPayload, out ChainedHeader lastHeader)
```

Comment:
This shouldn't be public. We don't want any external code to call it.

Reply:
Its made public so the test can call in to the code, I think its not a big deal if it is public.

Reply:
Imho it's exposes what shouldn't be exposed on public interface.
I think in general it's a bad practice to do changes that decrease quality of production code in favor of tests unless it's impossible to test something without doing that.

Reply:
Again, then the design of these classes aren't built in a very TDD way as it makes it VERY difficult to unit test critical logic in the code.

We either expose these methods or we create a `ConstructHeadersPayload` class that we can inject with `chain` and `provenBlockHeaderStore` so that we can easily new this up in a unit test and write test scenarios around it.